django/apps/existing_database/models.py

```diff
@@ -0,0 +1,213 @@
+    project_type = models.IntegerField(blank=True, null=True)
+    required_results = models.IntegerField(blank=True, null=True)
+    result_count = models.IntegerField(blank=True, null=True)
+    status = models.CharField(max_length=-1, blank=True, null=True)
+    verification_number = models.IntegerField(blank=True, null=True)
+    # Database uses JSON instead of JSONB (not supported by django)
+    project_type_specifics = models.TextField(blank=True, null=True)
+    organization_name = models.CharField(max_length=-1, null=True, blank=True)
+
+    class Meta:
+        managed = False
+        db_table = "projects"
+        unique_together = (("project_id", "organization"),)
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    def __str__(self):
+        return self.project_id
+
+
+class Group(models.Model):
+    # NOTE: Primary Key: project_id, group_id
+    project = models.ForeignKey("Project", models.DO_NOTHING, primary_key=True)
+    group_id = models.CharField(max_length=-1)
+    number_of_tasks = models.IntegerField(blank=True, null=True)
+    finished_count = models.IntegerField(blank=True, null=True)
+    required_count = models.IntegerField(blank=True, null=True)
+    progress = models.IntegerField(blank=True, null=True)
+    # Database uses JSON instead of JSONB (not supported by django)
+    project_type_specifics = models.TextField(blank=True, null=True)
+
+    class Meta:
+        managed = False
+        db_table = "groups"
+        unique_together = (("project", "group_id"),)
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    def __str__(self):
+        return f"P:{self.project_id}-G:{self.group_id}"
+
+
+class Task(models.Model):
+    # NOTE: Primary Key: project_id, group_id, tasks_id
+    project = models.ForeignKey(Project, models.DO_NOTHING, primary_key=True)
+    group_id = models.CharField(max_length=-1, primary_key=True)
+    task_id = models.CharField(max_length=-1, primary_key=True)
+    geom = gis_models.MultiPolygonField(blank=True, null=True)
+    # Database uses JSON instead of JSONB (not supported by django)
+    project_type_specifics = models.TextField(blank=True, null=True)
+
+    class Meta:
+        managed = False
+        db_table = "tasks"
+        unique_together = (("project", "group_id", "task_id"),)
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    @property
+    def group(self):
+        return Group.objects.filter(
+            project=self.project, group_id=self.group_id
+        ).first()
+
+    def __str__(self):
+        return f"P:{self.project_id}-G:{self.group_id}-T:{self.task_id}"
+
+
+class Result(models.Model):
+    # NOTE: Primary Key: project_id, group_id, tasks_id, user_id
+    project = models.ForeignKey(Project, models.DO_NOTHING, primary_key=True)
+    group_id = models.CharField(max_length=-1, primary_key=True)
+    task_id = models.CharField(max_length=-1, primary_key=True)
+    user = models.ForeignKey(User, models.DO_NOTHING, primary_key=True)
```

Comment:
here you define `project` and `user` based on ForeignKeys. Why are you not doing the same for group_id and task_id? In the current implementation this would mean that I can have a result, which defines a group_id which is not present in the groups table. Or do I miss something?

Reply:
Django needs to have a single primary key for maintaining relations. Due to multiple keys as primary in task and group, it isn't possible for now.